### PR TITLE
p4rt-2.1: fix usage of wantfail flag 

### PR DESCRIPTION
**Pull request by LimeHat**
This PR fixes the usage of wantFail flag for establishing p4rt stream channels; previously it was skipping error validation in some cases where it wasn't intended. 

---
<sup>
This code is a Contribution to the OpenConfig Feature Profiles project ("Work") made under the Google Software Grant and Corporate Contributor License Agreement ("CLA") and governed by the Apache License 2.0. No other rights or licenses in or to any of Nokia’s intellectual property are granted for any other purpose. This code is provided on an "as is" basis without any warranties of any kind.</sup>

### CODE REVIEW

**Comment by jjing1122** [comment]
filed internel bug b/277268421 for presubmit process

File: feature/experimental/p4rt/tests/p4rt_election/p4rt_election_test.go [hunk 1]
```diff
@@ -171,7 +170,7 @@ func getRespCode(args *testArgs) (int32, error) {
 }
 
 // getGDPParameter returns GDP related parameters for testPacketIn testcase.
-func getGDPParameter(t *testing.T) PacketIO {
+func getGDPParameter(_ *testing.T) PacketIO {
```
Comment (greg-dennis) [inline feature/experimental/p4rt/tests/p4rt_election/p4rt_election_test.go:173]: Can you just remove the parameter entirely and while you're at it, drop the word "get" from the method name to conform to go style? The name "gdpParameter()" is fine. There appears to only be one caller.

Comment (LimeHat) [reply feature/experimental/p4rt/tests/p4rt_election/p4rt_election_test.go]: I think we can remove this func altogether, moved it inline.

**Review by jjing1122 — APPROVED** [review]

**Pull request merged**